sound/soc/sof/topology.c

```diff
@@ -1445,6 +1445,53 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 	return 0;
 }
 
+/**
+ * sof_comp_alloc - alloc buffer for a new component and initialize it
+ * @swidget: SOF widget contained extend data
+ * @ipc_size: passed in with basic component size, will be updated to
+ *  the final IPC message size (including extended data) after returned
+ * @index: the pipeline ID the component belonged to
+ * @core: index of the DSP core that the component should run on
+ *
+ * Return: The pointer to the new allocated component, NULL if failed.
+ */
+static struct sof_ipc_comp *sof_comp_alloc(struct snd_sof_widget *swidget,
+					   size_t *ipc_size, int index,
+					   int core)
+{
+	struct sof_ipc_comp_ext nul_ext_data = {0};
+	struct sof_ipc_comp *comp;
+	size_t total_size = *ipc_size;
+
+	/* check if extended data valid */
+	if (memcmp(&swidget->comp_ext, &nul_ext_data,
+		   sizeof(swidget->comp_ext)))
```

Comment:
@keyonjie this will fit in one line with 100 chars no?

Reply:
> @keyonjie this will fit in one line with 100 chars no?

Yes, let me change it.